app/src/main/java/com/patrickgatewood/weather/injection/ApplicationModule.java

```diff
@@ -63,7 +64,7 @@ DarkSkyApi provideDarkskyApi(Retrofit retrofit) {
 
     @Provides
     @Singleton
-    WeatherViewModel provideWeatherViewModel(DarkSkyApi darkSkyApi) {
-        return new WeatherViewModel(darkSkyApi);
+    WeatherPresenter provideWeatherPresenter(DarkSkyApi darkSkyApi) {
```

Comment:
This might not always be a singleton. If the view is no longer valid, you may not want the presenter to keep its state.